### ISSUE TITLE
How can I visualize just like the demo video from my own video or in real-time?

### ISSUE THREAD

**Issue by kingsman0000** [facebookresearch/InterHand2.6M#7]

**Comment by mks0601** [comment]
sorry visualizing on other videos will be supported later.

**Comment by lan786** [comment]
@mks0601 Is there any update about this issue (real-time testing)?

**Comment by mks0601** [comment]
Sorry I'll be very busy until CVPR deadline (mid of Nov.) :(

**Comment by mks0601** [comment]
Hi guys. I added a demo code for a random image.

**Comment by shen494157765** [comment]
Hi, can you share which model you used in that demo for the given picture? 
the InterHand H or M or H+M?

**Comment by mks0601** [comment]
The link to the pre-trained model in the demo section is trained on H+M. Basically, H+M is always better than H and M.

**Comment by shen494157765** [comment]
> The link to the pre-trained model in the demo section is trained on H+M. Basically, H+M is always better than H and M.

Thanks a lot for your response!

**Comment by shen494157765** [comment]
> The link to the pre-trained model in the demo section is trained on H+M. Basically, H+M is always better than H and M.

Hi, I have used the model based on interhand dataset with H+M on google drive, but I did not get the same result with you, I mean the picture you gave in demo file named 'result_2d.jpg'.

Is there anything wrong with the parameter of bbox? I just used the bbox = [69, 137, 165, 153]

By the way, what I got is below:
![git_result](https://user-images.githubusercontent.com/39080134/102825466-ceece680-43ac-11eb-9aed-301b745a99fb.jpg)

**Comment by mks0601** [comment]
That's weird... I got the same result like below.
![result_2d](https://user-images.githubusercontent.com/13601723/102830673-3480b780-442d-11eb-826d-1d71d8daebff.jpg)

Could you add `bbox = np.array(bbox, dtype=np.float32)` right after `bbox = [69, 137, 165, 153]`?

**Comment by shen494157765** [comment]
> That's weird... I got the same result like below.
> ![result_2d](https://user-images.githubusercontent.com/13601723/102830673-3480b780-442d-11eb-826d-1d71d8daebff.jpg)
> 
> Could you add `bbox = np.array(bbox, dtype=np.float32)` right after `bbox = [69, 137, 165, 153]`?

Hi, the result is same as I showed above.

Anyway, it is not important, it might be caused by some attribution of the picture.

By the way, I have converted your picture demo code into a video (camera) demo, and I have realized it in real-time. What I found is that, the model is good for the right hand, but for left hand, it is bad. Pictures below are some photos of my experiments.

![image](https://user-images.githubusercontent.com/39080134/102906700-ff7f5f80-4442-11eb-8673-1e42ce8bea84.png)
![image](https://user-images.githubusercontent.com/39080134/102906713-05754080-4443-11eb-9092-44a379487f9a.png)
![image](https://user-images.githubusercontent.com/39080134/102906737-0efea880-4443-11eb-8e7b-2879563d3e4e.png)
![image](https://user-images.githubusercontent.com/39080134/102906758-16be4d00-4443-11eb-9473-156d958e964d.png)
![image](https://user-images.githubusercontent.com/39080134/102906775-1d4cc480-4443-11eb-81bf-1f7e50eb1551.png)

If you have any recommendations, please let me know.

thanks for your response again!

**Comment by mks0601** [comment]
This is weird. The accuracy of the right and left hand are almost the same. Could you tell me how did you change the code and could you provide your demo image?

**Comment by shen494157765** [comment]
> This is weird. The accuracy of the right and left hand are almost the same. Could you tell me how did you change the code and could you provide your demo image?

my camera demo code is not really different from your picture demo code, I just change the bbox into the whole pic, I mean: 

original_img_height, original_img_width = original_img.shape[:2]
bbox = [0, 0, original_img_height, original_img_width]

The reason is that I found it can bring better results when I put my hand in front of the camera and when the hand take the most area of the video (I mean when the background is white and the whole camera is filled with hand) it can get good results.

By the way, I just inference the picture and draw the picture with every frame of the camera, and the only parameter I changed was the **bbox**

**Comment by mks0601** [comment]
I guess the reason is `bbox`. You should crop the hand area or if you want to set `bbox` as the whole image area, your hand must be close to the camera.

**Comment by shen494157765** [comment]
> I guess the reason is `bbox`. You should crop the hand area or if you want to set `bbox` as the whole image area, your hand must be close to the camera.

Yeah, I know that, but I am considering the reason that why it can get good result only with right hand....

**Comment by shen494157765** [comment]
> I guess the reason is `bbox`. You should crop the hand area or if you want to set `bbox` as the whole image area, your hand must be close to the camera.

Hi, I found where the problem is, I just forget that I have changed some parts of the net, but in the mean time, I still use your model ( ><,  the code even not got an error....).

Now I have got the same demo result with you.

Thanks for your patience!
Hope you safe and healthy!

**Comment by mks0601** [comment]
Yes I disabled 'strict' option when loading the pre-trained weights (see [here](https://github.com/facebookresearch/InterHand2.6M/blob/b32ff133ffbdec4f123a1ccfe6b2195bf950a508/common/base.py#L152)). I think I did this for the debugging purpose. Let me enable this option. Thanks!

**Comment by Algabeno** [comment]
@shen494157765 Hi,Can you share with me the camera demo code

**Comment by vodanhbk95** [comment]
> > I guess the reason is `bbox`. You should crop the hand area or if you want to set `bbox` as the whole image area, your hand must be close to the camera.
> 
> Hi, I found where the problem is, I just forget that I have changed some parts of the net, but in the mean time, I still use your model ( ><, the code even not got an error when loading the weights....).
> 
> Now I have got the same demo result with you.
> 
> Thanks for your patience!
> Hope you safe and healthy!

Hi @shen494157765 , Can you tell me more about where the problem is and what you have fixed it. 
I am do the same thing like you:
original_img_height, original_img_width = original_img.shape[:2]
bbox = [0, 0, original_img_height, original_img_width]
It got bad result. So please share me more.

**Comment by shen494157765** [comment]
> > > I guess the reason is `bbox`. You should crop the hand area or if you want to set `bbox` as the whole image area, your hand must be close to the camera.
> > 
> > 
> > Hi, I found where the problem is, I just forget that I have changed some parts of the net, but in the mean time, I still use your model ( ><, the code even not got an error when loading the weights....).
> > Now I have got the same demo result with you.
> > Thanks for your patience!
> > Hope you safe and healthy!
> 
> Hi @shen494157765 , Can you tell me more about where the problem is and what you have fixed it.
> I am do the same thing like you:
> original_img_height, original_img_width = original_img.shape[:2]
> bbox = [0, 0, original_img_height, original_img_width]
> It got bad result. So please share me more.

Hi, the problem I had was not caused by the bbox, the problem was caused by the different model.
As I said, I revised the original model, which means my model was different with the author's, but the old code still allows the revised model load the weights from original model.

**Comment by shen494157765** [comment]
> @shen494157765 Hi,Can you share with me the camera demo code

It is easy to write, what you need to do is just show every frame which is the demo picture result

**Comment by ZhengdiYu** [comment]
> > @shen494157765 Hi,Can you share with me the camera demo code
> 
> It is easy to write, what you need to do is just show every frame which is the demo picture result

Interesting. Did you split the network into two identical branches? I was wondering why would this happen. Because normally if you disable some part of the network, both two hands should be degenerated at the same time.
